From 2b2539f7542aca08cbdd58b6c2adf0fbfdc53648 Mon Sep 17 00:00:00 2001
From: Jukka Rissanen <jukka.rissanen@linux.intel.com>
Date: Wed, 23 Jul 2014 10:47:59 +0300
Subject: [PATCH 3/6] kernel: Use KERNEL_MODULE_AUTOLOAD

The module_autoload_xxx is deprecated so use KERNEL_MODULE_AUTOLOAD
---
 .../linux/linux-yocto-quark_3.8.bbappend           | 11 ++++----
 .../recipes-kernel/linux/linux-yocto-quark_3.8.bb  | 29 +++++++++++-----------
 2 files changed, 20 insertions(+), 20 deletions(-)

diff --git a/meta-iot-devkit/recipes-kernel/linux/linux-yocto-quark_3.8.bbappend b/meta-iot-devkit/recipes-kernel/linux/linux-yocto-quark_3.8.bbappend
index fa34414..a0e468e 100644
--- a/meta-iot-devkit/recipes-kernel/linux/linux-yocto-quark_3.8.bbappend
+++ b/meta-iot-devkit/recipes-kernel/linux/linux-yocto-quark_3.8.bbappend
@@ -1,13 +1,14 @@
-module_autoload_iwlwifi = "iwlwifi"
-module_autoload_btusb = "btusb"
+KERNEL_MODULE_AUTOLOAD += "iwlwifi"
+KERNEL_MODULE_AUTOLOAD += "btusb"
 
 # swap g_serial for g_acm_ms
-module_autoload_pch_udc = " pch_udc g_acm_ms"
+KERNEL_MODULE_AUTOLOAD += "pch_udc g_acm_ms"
+KERNEL_MODULE_PROBECONF += "g_acm_ms"
 module_conf_g_acm_ms = "options g_acm_ms file=/dev/mmcblk0p1 removable=1 idVendor=0x8086 idProduct=0xDEAD"
 
 # PPP module autoload
-module_autoload_pppox = "pppox"
-module_autoload_pppoe = "pppoe"
+KERNEL_MODULE_AUTOLOAD += "pppox"
+KERNEL_MODULE_AUTOLOAD += "pppoe"
 
 # find defconfig path
 FILESEXTRAPATHS := "${THISDIR}/${PN}/:${THISDIR}/${PN}/iot-devkit/"
diff --git a/meta-quark-bsp/recipes-kernel/linux/linux-yocto-quark_3.8.bb b/meta-quark-bsp/recipes-kernel/linux/linux-yocto-quark_3.8.bb
index c9bfd39..0e9956a 100644
--- a/meta-quark-bsp/recipes-kernel/linux/linux-yocto-quark_3.8.bb
+++ b/meta-quark-bsp/recipes-kernel/linux/linux-yocto-quark_3.8.bb
@@ -102,25 +102,24 @@ SRC_URI += "file://uart-1.0.patch"
 # character seperator.
 
 # USB Host
-module_autoload_ehci-hcd = "ehci-hcd"
-module_autoload_ehci-pci = "ehci-pci"
-module_autoload_ohci-hcd = "ohci-hcd"
-module_autoload_usb-storage = "usb-storage"
-module_autoload_usbhid = "usbhid"
-module_autoload_evdev = "evdev"
+KERNEL_MODULE_AUTOLOAD += "ehci-hcd ehci-pci ohci-hcd usb-storage usbhid evdev"
+
 # USB Device (pch_udc is required for g_serial to load)
-module_autoload_pch_udc = "pch_udc g_serial"
+KERNEL_MODULE_AUTOLOAD += "pch_udc g_serial"
+KERNEL_MODULE_PROBECONF += "g_serial"
 module_conf_g_serial = "options g_serial vendor=0x8086 product=0xBABE"
+
 # SDHC
-module_autoload_sdhci-pci = "sdhci-pci"
-module_autoload_mmc-block = "mmc_block"
+KERNEL_MODULE_AUTOLOAD += "sdhci-pci mmc_block"
+
 # SPI
-module_autoload_spidev = "spidev"
-module_autoload_spi-pxa2xx = "spi-pxa2xx"
-module_autoload_spi-pxa2xx-pci = "spi-pxa2xx-pci"
+KERNEL_MODULE_AUTOLOAD += "spidev spi-pxa2xx spi-pxa2xx-pc"
+
 # GPIO
-module_autoload_gpio-sch = "gpio-sch"
+KERNEL_MODULE_AUTOLOAD += "gpio-sch"
+
 # Ethernet
-module_autoload_stmmac = "stmmac"
+KERNEL_MODULE_AUTOLOAD += "stmmac"
+
 # EEPROM Access
-module_autoload_at24 = "at24"
+KERNEL_MODULE_AUTOLOAD += "at24"
-- 
1.8.3.1

